### PR TITLE
Disable the SimplifyArmIdentity mir-opt

### PR DESCRIPTION
**Pull request by wesleywiser**
The optimization still has some bugs that need to be worked out
such as #77359.

We can try re-enabling this again after the known issues are resolved.

r? @oli-obk

### CODE REVIEW

File: src/test/mir-opt/simplify_locals_removes_unused_discriminant_reads.map.SimplifyLocals.64bit.diff [hunk 1]
```diff
@@ -2,25 +2,25 @@
 + // MIR for `map` after SimplifyLocals
   
   fn map(_1: Option<Box<()>>) -> Option<Box<()>> {
-      debug x => _1;                       // in scope 0 at $DIR/simplify-locals-removes-unused-discriminant-reads.rs:1:8: 1:9
```
Comment (oli-obk) [inline src/test/mir-opt/simplify_locals_removes_unused_discriminant_reads.map.SimplifyLocals.64bit.diff:5]: you also need to update the 32 bit version of this file

**Comment by oli-obk** [comment]
r=me with 32 bit files updated

File: src/test/mir-opt/simplify_locals_removes_unused_discriminant_reads.map.SimplifyLocals.64bit.diff (hunk 1 above)
Comment (wesleywiser) [reply src/test/mir-opt/simplify_locals_removes_unused_discriminant_reads.map.SimplifyLocals.64bit.diff:5]: Oh, duh 🤦

**Comment by wesleywiser** [comment]
@bors r=oli-obk rollup=never p=5

Closes an unsound nightly regression.

**Pull request merged**